File: server/spring-boot-ecommerce/src/main/java/com/ecommerce/springbootecommerce/dto/OrderRequest.java
package com.ecommerce.springbootecommerce.dto;

import com.ecommerce.springbootecommerce.domain.Address;
import com.ecommerce.springbootecommerce.domain.Customer;
import com.ecommerce.springbootecommerce.domain.OrderItem;
import lombok.Data;

import javax.validation.constraints.NotNull;
import java.util.Set;

@Data
public class OrderRequest {

    @NotNull(message = "customer cannot be empty")
    private Customer customer;

    @NotNull(message = "shipping address cannot be empty")
    private Address shippingAddress;

    @NotNull(message = "order cannot be empty")
    private OrderDTO orderDTO;

    @NotNull(message = "order items cannot be empty")
    private Set<OrderItem> orderItems;
}

File: client/ecommerce-app/src/app/modules/models/orderRequest.ts
import { Address } from './address';
import { Order } from './order';
import { OrderItem } from './orderItem';
import { User } from './user';

export class OrderRequest {
  user!: User;
  shippingAddress!: Address;
  order!: Order;
  orderItems!: OrderItem[];
}
